src/services/node-checker-wrapper.ts

```diff
@@ -0,0 +1,272 @@
+import { Redis } from 'ioredis'
+import { Pocket, Node, PocketAAT, Configuration, Session } from '@pokt-network/pocket-js'
+import { getNodeNetworkData, removeNodeFromSession } from '../utils/cache'
+import { MAX_RELAYS_ERROR } from '../utils/constants'
+import { hashBlockchainNodes } from '../utils/helpers'
+import { MetricsRecorder } from './metrics-recorder'
+import { ArchivalCheck, ChainCheck, Check, NodeChecker, NodeCheckResponse, SyncCheck } from './node-checker'
+
+const logger = require('../services/logger')
+
+export class NodeCheckerWrapper {
+  pocket: Pocket
+  redis: Redis
+  metricsRecorder: MetricsRecorder
+  origin: string
+
+  constructor(pocket: Pocket, redis: Redis, metricsRecorder: MetricsRecorder, origin: string) {
+    this.pocket = pocket
+    this.redis = redis
+    this.metricsRecorder = metricsRecorder
+    this.origin = origin
+  }
+
+  /**
+   * Helper method to check for cached checks, is a check is already cached or in progress,
+   * returns valid nodes from the cached result or session. Otherwise set a cache lock.
+   * @param {Node[]} nodes session nodes.
+   * @param cacheKey key to get/set results.
+   * @returns nodes cached or provided on case of cache/lock.
+   */
+  protected async cacheNodes(nodes: Node[], cacheKey: string): Promise<Node[]> {
```

Comment:
This function doesn't cache nodes; it checks if they're cached and retrieves them or sets a cache lock. I think this function name could be a little bit better!